app/wasmtest/helpers_test.go

```diff
@@ -0,0 +1,45 @@
+package wasmtest
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/osmosis-labs/osmosis/app"
+)
+
+func CreateTestInput() (*app.OsmosisApp, sdk.Context) {
+	osmosis := app.Setup(false)
+	ctx := osmosis.BaseApp.NewContext(false, tmproto.Header{Height: 1, ChainID: "osmosis-1", Time: time.Now().UTC()})
+	return osmosis, ctx
+}
+
+func FundAccount(t *testing.T, ctx sdk.Context, osmosis *app.OsmosisApp, acct sdk.AccAddress) {
+	err := simapp.FundAccount(osmosis.BankKeeper, ctx, acct, sdk.NewCoins(
+		sdk.NewCoin("uosmo", sdk.NewInt(10000000000)),
+	))
+	require.NoError(t, err)
+}
+
+// we need to make this deterministic (same every test run), as content might affect gas costs
+func keyPubAddr() (crypto.PrivKey, crypto.PubKey, sdk.AccAddress) {
+	key := ed25519.GenPrivKey()
```

Comment:
So do we need to switch to GenPrivKeyFromSecret? (https://github.com/tendermint/tendermint/blob/master/crypto/ed25519/ed25519.go#L144 )

Also a bit surprised that this is an ed25519 address, versus a secp one.

Reply:
Yeah, copied some code I found elsewhere (in wasmd).

The original one in wasmd had a deterministic seed (generated from a counter). I tried making this random, but can use this. 

Can you link me to where you generate random addresses in your codebase? I can just import that instead

Reply:
Or just update this code in a separate PR yourself. As long as it returns valid addresses, it is fine